# A plotting R script produced by the Revigo server at http://revigo.irb.hr/
# If you found Revigo useful in your work, please cite the following reference:
# Supek F et al. "REVIGO summarizes and visualizes long lists of Gene Ontology
# terms" PLoS ONE 2011. doi:10.1371/journal.pone.0021800

# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );

# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );

# --------------------------------------------------------------------------
# Here is your data from Revigo. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency","plot_X","plot_Y","log_size","value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0000097","sulfur amino acid biosynthetic process",0.4753877262335075,-4.6411328384160155,5.415870161676681,5.15510516225765,-1.2523551806717517,0.5995476219952025,0.00721518),
c("GO:0006091","generation of precursor metabolites and energy",2.62835465282124,0.1463274098147802,-1.883231210076417,5.897738674144431,-0.3678465164893673,0.8945710408009726,0.09500459),
c("GO:0006334","nucleosome assembly",0.09057190085107651,3.1627324443350857,6.261950655519788,4.435063590165067,-0.7533266533060027,0.8496725491722773,0.54216651),
c("GO:0006414","translational elongation",0.45853729473473576,-5.597653241499517,4.329224295863344,5.139431973597713,-1.0076586004423747,0.7049380999222833,0.30565017),
c("GO:0006457","protein folding",1.0590625918025878,6.568219919317553,2.8525018316134423,5.502977151016975,-1.239645380391584,0.9832990491286956,0.00987736),
c("GO:0006468","protein phosphorylation",4.340892012483024,-7.02910547733115,1.6453603548222178,6.115633473764006,-0.6645802145866276,0.76144486348302,0.32027556),
c("GO:0006790","sulfur compound metabolic process",1.9505422206753265,2.3497671785455116,0.9400410488009424,5.768210254776623,-0.5898860990864722,0.8987993527417556,0.07750958),
c("GO:0006811","monoatomic ion transport",4.401907471506606,3.136837257465785,-4.669983338674411,6.1216953964406695,-0.6543107118234198,0.941194935154322,0.41209348),
c("GO:0007021","tubulin complex assembly",0.022242170436692937,3.5420010692367807,6.03504211529399,3.82529626443096,-1.3222197290286184,0.8700754437962397,0.00588505),
c("GO:0007023","post-chaperonin tubulin folding pathway",0.02748423124246953,6.7209330463670955,0.33642286106733177,3.917190308511564,-1.5797801222748522,0.9883465375969931,0),
c("GO:0007154","cell communication",8.661893464294382,5.337255232841072,-2.965010107120036,6.415667163528361,-0.41795255393165576,0.9777023434581711,0.01624466),
c("GO:0007165","signal transduction",8.135788134528843,-1.277921384934595,-5.851443510435818,6.388453966952047,-0.42372086088641503,0.9389088431644134,0.46662669),
c("GO:0009059","macromolecule biosynthetic process",15.258385035486357,-5.1743371018143165,2.593295165399621,6.661562823384344,-0.5482565356668755,0.717505335021589,0.50121883),
c("GO:0009066","aspartate family amino acid metabolic process",0.8335242561063357,-4.351614762560542,7.154684335990721,5.398974134528076,-0.5509074688805811,0.6828557036837142,0.67656516),
c("GO:0009987","cellular process",79.12519251933975,5.992678149690259,-1.350649708896036,7.376368956838209,-0.3211184950379713,1,-0),
c("GO:0010467","gene expression",9.706370753664654,-8.07420050979701,-0.10105178023703829,6.465111183714701,-0.7100544838233321,0.8884579374742145,0.12284429),
c("GO:0015980","energy derivation by oxidation of organic compounds",1.3592204656440126,3.16301391053335,-1.1466879912389956,5.611345126683129,-0.4238752750649395,0.9034806936469275,0.08739706),
c("GO:0016310","phosphorylation",7.474579996508841,-6.97282605756628,-2.3165799529051676,6.35164114641232,-0.4726227593113076,0.8592448028760719,0.62371761),
c("GO:0019538","protein metabolic process",18.67385374490695,-6.638026648237194,2.358326750835292,6.7492881892860686,-1.1492680169674065,0.7597312351120995,0.35014548),
c("GO:0022607","cellular component assembly",2.5732697740112993,2.7596984329380105,6.75152108939418,5.888540038987443,-1.1937521728042095,0.840316708311187,0.47909884),
c("GO:0023052","signaling",8.271383217529026,-1.6065943168349166,-5.6534976319059185,6.395632475900429,-0.545118569364387,0.9604941439943318,0.30934204),
c("GO:0034220","monoatomic ion transmembrane transport",3.707494071415495,2.736800206737558,-4.723143976651222,6.047135013730295,-0.5662429631544852,0.8901333509307787,0.50368514),
c("GO:0034728","nucleosome organization",0.09438037776897892,3.417532592071607,6.465243657052254,4.452951175463887,-0.7269987279362623,0.8556511486571943,0.67893359),
c("GO:0043170","macromolecule metabolic process",38.34861513800723,-3.9952983121738623,-2.001131891937731,7.061803881678349,-0.5220139132762343,0.884594580457122,0.20088823),
c("GO:0043603","amide metabolic process",6.518961624921768,-5.696496373962394,-3.403299418233348,6.292232804957447,-1.0281769880886455,0.8905321416822108,0.18637799),
c("GO:0043604","amide biosynthetic process",5.857400911746049,-4.082600759440649,3.323688207017921,6.245759355967277,-1.2803455988016106,0.6653865894233827,0.6208586),
c("GO:0043933","protein-containing complex organization",2.6611873826523444,2.689565507968734,7.053910759468522,5.903130157373668,-0.5311743228728963,0.8546886576391051,0.64626874),
c("GO:0044085","cellular component biogenesis",4.796572117984153,2.4429117632781905,7.0922579126003535,6.158985438645769,-0.7304006724606216,0.8563394713727436,0.63827793),
c("GO:0044249","cellular biosynthetic process",24.389190152163454,-3.133047442395381,3.2696135093890937,6.865251597822689,-0.49134775320514945,0.6942098573988112,0.55144154),
c("GO:0044271","cellular nitrogen compound biosynthetic process",18.93399433857433,-4.034561401165288,2.384960839690856,6.755296478091295,-0.3995041104950825,0.6722755106538908,0.59014338),
c("GO:0044283","small molecule biosynthetic process",6.034460163531009,-3.740129930580656,5.727514588837274,6.258692822898982,-0.3486929047557773,0.6928885664488316,0.5211708),
c("GO:0050790","regulation of catalytic activity",0.6743565303837294,-2.6439364969378936,-5.7670229495422864,5.306945868486873,-1.1459856104446353,0.969405037046555,-0),
c("GO:0065004","protein-DNA complex assembly",0.1970396193359134,3.120990604877015,6.562986922844697,4.772615049849171,-0.6425606836861009,0.8427240861384064,0.66833899),
c("GO:0065009","regulation of molecular function",0.8166305842582776,-1.9457688979225305,-6.043422686594042,5.3900815687627786,-1.0300113086427087,0.9688299161291769,0.23026939),
c("GO:0071840","cellular component organization or biogenesis",9.440442605554349,5.449029173074584,1.4530547882774691,6.453046668121093,-0.30258058599369564,0.9773923272725663,0.02033788),
c("GO:0098660","inorganic ion transmembrane transport",2.8120629398541377,2.6304787108875396,-5.01989238195356,5.9270797094445795,-0.7403058006432529,0.9167452569592176,0.01203986),
c("GO:1901564","organonitrogen compound metabolic process",29.993834591427927,-5.2388038477361,-0.30182490672080664,6.955086199151694,-1.1004202312483935,0.8439855316288687,0.11190784),
c("GO:1901566","organonitrogen compound biosynthetic process",14.63667861962968,-4.915852076671693,3.4917767788523824,6.643496794616454,-0.9668940391286854,0.6800252890344701,0.46837375),
c("GO:1901576","organic substance biosynthetic process",27.07678740968754,-3.5615237379895452,2.927812397679228,6.910651347384342,-0.4558835655319974,0.709039150726502,0.66782618),
c("GO:1901605","alpha-amino acid metabolic process",3.5585210895822956,-4.957090540051152,6.278096919784012,6.029324108830442,-0.5118837952735734,0.6411975854505033,0.51588056));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$log_size <- as.numeric( as.character(one.data$log_size) );
one.data$value <- as.numeric( as.character(one.data$value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = value, size = log_size), alpha = I(0.6) );
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = log_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) ));
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ];
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);


# --------------------------------------------------------------------------
# Output the plot to screen
pdf(file="H_C_VS_T2_scatterplot.pdf", width=16, height=9 ) 
p1
dev.off()

